src/main/scala/firrtl/transforms/DeadCodeElimination.scala

```diff
@@ -0,0 +1,251 @@
+  def inputForm = LowForm
+  def outputForm = LowForm
+
+  /** Based on LogicNode ins CheckCombLoops, currently kind of faking it */
+  private type LogicNode = MemoizedHash[WrappedExpression]
+  private object LogicNode {
+    def apply(moduleName: String, expr: Expression): LogicNode =
+      WrappedExpression(Utils.mergeRef(WRef(moduleName), expr))
+    def apply(moduleName: String, name: String): LogicNode = apply(moduleName, WRef(name))
+    def apply(component: ComponentName): LogicNode = {
+      // Currently only leaf nodes are supported TODO implement
+      val loweredName = LowerTypes.loweredName(component.name.split('.'))
+      apply(component.module.name, WRef(loweredName))
+    }
+  }
+
+  /** Expression used to represent outputs in the circuit (# is illegal in names) */
+  private val circuitSink = LogicNode("#Top", "#Sink")
+
+  /** Extract all References and SubFields from a possibly nested Expression */
+  def extractRefs(expr: Expression): Seq[Expression] = {
+    val refs = mutable.ArrayBuffer.empty[Expression]
+    def rec(e: Expression): Expression = {
+      e match {
+        case ref @ (_: WRef | _: WSubField) => refs += ref
+        case nested @ (_: Mux | _: DoPrim | _: ValidIf) => nested map rec
+        case ignore @ (_: Literal) => // Do nothing
+        case unexpected => throwInternalError
+      }
+      e
+    }
+    rec(expr)
+    refs
+  }
+
+  /** Construct the dependency graph within this module */
+  private def setupDepGraph(depGraph: MutableDiGraph[LogicNode],
+                            instMap: collection.Map[String, String])
+                           (mod: Module): Unit = {
+    // Gets all dependencies and constructs LogicNodes from them
+    def getDeps(expr: Expression): Seq[LogicNode] =
+      extractRefs(expr).map { e =>
+        if (kind(e) == InstanceKind) {
+          val (inst, tail) = Utils.splitRef(e)
+          LogicNode(instMap(inst.name), tail)
+        } else {
+          LogicNode(mod.name, e)
+        }
+      }
+
+    def onStmt(stmt: Statement): Unit = stmt match {
+      case DefRegister(_, name, _, clock, reset, init) =>
+        val node = LogicNode(mod.name, name)
+        depGraph.addVertex(node)
+        Seq(clock, reset, init).flatMap(getDeps(_)).foreach(ref => depGraph.addEdge(node, ref))
+      case DefNode(_, name, value) =>
+        val node = LogicNode(mod.name, name)
+        depGraph.addVertex(node)
+        getDeps(value).foreach(ref => depGraph.addEdge(node, ref))
+      case DefWire(_, name, _) =>
+        depGraph.addVertex(LogicNode(mod.name, name))
+      case mem: DefMemory =>
+        // Treat DefMems as a node with outputs depending on the node and node depending on inputs
+				// From perpsective of the module or instance, MALE expressions are inputs, FEMALE are outputs
+        val memRef = WRef(mem.name, MemPortUtils.memType(mem), ExpKind, FEMALE)
+				val exprs = Utils.create_exps(memRef).groupBy(Utils.gender(_))
```

Comment:
Sometimes when I turn paste on to copy paste something in I forget to turn it off and weird stuff like this happens. Good catch.